src/engine/MessageQueue.inl.hxx

```diff
@@ -13,16 +13,30 @@ template <typename Event> void MessageQueue<Event>::push(Event &&event)
 template <typename Event> bool MessageQueue<Event>::peek(void) noexcept
 {
   Lock<Mutex> lock(*m_Semaphore);
```

Comment:
Can you rename `m_Semaphore` to `m_Mutex`? I wrote this code before I could understand what it meant lol